controllers/machineselector_controller.go

```diff
@@ -594,25 +596,34 @@ func (r *MachineInventorySelectorReconciler) setInvetorySelectorAddresses(ctx co
 	return nil
 }
 
-func (r *MachineInventorySelectorReconciler) ignoreIncrementalStatusUpdate() predicate.Funcs {
+func filterSelectorUpdateEvents() predicate.Funcs {
 	return predicate.Funcs{
-		// Avoid reconciling if the event triggering the reconciliation is related to incremental status updates
-		// for MachineInventorySelector resources only
+
 		UpdateFunc: func(e event.UpdateEvent) bool {
-			if e.ObjectOld.GetObjectKind().GroupVersionKind().Kind != "MachineInventorySelector" {
-				return true
+			switch e.ObjectOld.GetObjectKind().GroupVersionKind().Kind {
+			case "MachineInventorySelector":
+				// Avoid reconciling if the event triggering the reconciliation is related to
+				// incremental status updates for MachineInventorySelector resources only
+				oldS := e.ObjectOld.(*elementalv1.MachineInventorySelector).DeepCopy()
+				newS := e.ObjectNew.(*elementalv1.MachineInventorySelector).DeepCopy()
+
+				oldS.Status = elementalv1.MachineInventorySelectorStatus{}
+				newS.Status = elementalv1.MachineInventorySelectorStatus{}
+
+				oldS.ObjectMeta.ResourceVersion = ""
+				newS.ObjectMeta.ResourceVersion = ""
+				return !cmp.Equal(oldS, newS)
+			case "MachineInventory":
+				// Avoid reconciling if the event triggering the reconciliation is related to
+				// adopting an inventory resource
+				oldI := e.ObjectOld.(*elementalv1.MachineInventory)
+				newI := e.ObjectNew.(*elementalv1.MachineInventory)
+
+				// Do not reconcile if new owners are set
+				return len(oldI.ObjectMeta.OwnerReferences) >= len(newI.ObjectMeta.OwnerReferences)
```

Comment:
This only omitting the reconcile loop in case a new owner is added, this is to prevent selectors re-triggering themself when adding an owner to inventories.